orc8r/tools/fab/vagrant.py

```diff
@@ -94,16 +75,41 @@ def setup_env_vagrant(machine='magma', apply_to_env=True, force_provision=False)
         }
 
 
+def _is_vm_running(machine):
+    return local(f"vagrant status {machine}", capture=True) \
+        .find('running') >= 0
+
+
+def _vagrant_up_with_retry(machine, max_retries):
+    print(f"VM {machine} is not running - attempting to bring it up ...")
+    for attempt in range(max_retries):
+        print(f"... attempt {attempt + 1}/{max_retries} to bring up VM {machine}")
+
+        try:
+            local(f"vagrant up {machine}")
+        except:
+            print(f"... VM {machine} failed during 'vagrant up' - VM will be destroyed")
+            teardown_vagrant(machine)
+
+        if _is_vm_running(machine):
+            print(f"... VM {machine} is running.")
+            break  # success
+
+    if not _is_vm_running(machine):
+        print(f"Error: VM: {machine} is still not running after {max_retries} attempt(s).")
```

Comment:
```suggestion
        print(f"Error: VM {machine} is still not running after {max_retries} attempt(s).")
```

Reply:
Small suggestion for consistency with the above formatting

Reply:
:+1: changed and rebased